.github/RELEASE.md

```diff
@@ -2,12 +2,16 @@
 
 ## Release Process
 
+> **Note:**
+> Throughout this document are references to the version number as `YY.M.0`, this should be replaced with the correct version number. Do **not** prefix the version with a lowercase `v`.
```

Comment:
Good call-out of the "v" prefix not being needed!

Should we also link to CEP 8 here?

Reply:
Hm, I hadn't included it since this is synced out to more than just conda but it won't hurt to mention it.

Reply:
Decided to leave this out for now since CEP 8 is not applicable for all repos.